File: solutions/python/regression_foundations_ex1.py

import numpy as np
import pandas as pd
import statsmodels.api as sm
import matplotlib.pyplot as plt

np.random.seed(42)
n = 150
age = np.random.uniform(30, 75, n).round()
sbp = 85 + 0.6 * age + np.random.normal(0, 12, n)

clinical_data = pd.DataFrame({'age': age, 'sbp': sbp})

# Fit the model
X = sm.add_constant(clinical_data['age'])
model = sm.OLS(clinical_data['sbp'], X).fit()
print(model.summary())

print(f"\nIntercept: {model.params['const']:.2f} mmHg")
print(f"Slope: {model.params['age']:.3f} mmHg per year of age")
print(f"R-squared: {model.rsquared:.3f}")
print(f"95% CI for slope: ({model.conf_int().loc['age', 0]:.3f}, "
      f"{model.conf_int().loc['age', 1]:.3f})")

# Scatterplot with regression line
fig, ax = plt.subplots(figsize=(8, 6))
ax.scatter(age, sbp, alpha=0.5)
age_range = np.linspace(30, 75, 100)
predicted = model.params['const'] + model.params['age'] * age_range
ax.plot(age_range, predicted, 'b-', linewidth=2, label='Regression line')
ax.set_xlabel('Age (years)')
ax.set_ylabel('Systolic Blood Pressure (mmHg)')
ax.set_title('Age vs. Systolic Blood Pressure')
ax.legend()
plt.tight_layout()
plt.show()

File: solutions/R/regression_foundations_ex1.R

# Simulate clinical data
set.seed(42)
n <- 150
age <- round(runif(n, 30, 75))
sbp <- 85 + 0.6 * age + rnorm(n, 0, 12)

clinical_data <- data.frame(age = age, sbp = sbp)

# Fit the model
model <- lm(sbp ~ age, data = clinical_data)
summary(model)

# Interpret
cat("\nIntercept:", round(coef(model)[1], 3), "mmHg\n")
cat("Slope:", round(coef(model)[2], 3), "mmHg per year of age\n")
cat("R-squared:", round(summary(model)$r.squared, 3), "\n")

# Confidence interval for the slope
confint(model, "age", level = 0.95)

# Scatterplot with regression line
library(ggplot2)
ggplot(clinical_data, aes(x = age, y = sbp)) +
  geom_point(alpha = 0.5) +
  geom_smooth(method = "lm", se = TRUE, color = "blue") +
  labs(
    x = "Age (years)",
    y = "Systolic Blood Pressure (mmHg)",
    title = "Age vs. Systolic Blood Pressure"
  ) +
  theme_minimal()
